specs/workload/Workload.md

```diff
@@ -0,0 +1,168 @@
+
+* Id -- programmatic identifier
+* DisplayName -- localizable name for user/human consumption
+* Handler -- WinRT object providing operations manipulating the workload
+
+An MSIX package can define one or more workloads. The following manifest schema is used:
+
+```xml
+<Package>
+  ...
+  <Properties>
+    <workload:Workloads>
+      <Workload Id="id" DisplayName="dn">
+        <Actions>
+          <Handler ActivatableClassId="acid">
+        </Actions>
+      </Workload>
+      ...
+    </Workload:Workloads>
+  </Properties>
+  ...
+</Package>
+```
+
+The following validations apply:
+
+* Id (id)
+  * String length: 1-255
+  * Allowed characters: [A-Za-z0-9_.]
+  * Comparison: Case-Insensitive
+  * Occurence: Unique across all workloads in a package
+* DisplayName (dn)
+  * Standard `DisplayName` definition
+* ActivatableClassId (acid)
+  * Standard `ActivatableClassId` definition
+
+# 4. Examples
+
+## 4.1. Remove all workloads
+
+Fabrikam app removing all workloads.
+
+```c#
+namespace RemoveAllWorkloads
+{
+    static class Program
+    {
+        static void Main()
+        {
+            var workloadManager = WorkloadManager.GetDefault();
+            foreach (string id in workloadManager.FindIds())
+            {
+                var workload = workloadManager.GetWorkload(id)
+                Console.Write($"Removing {workload.DisplayName}...");
+                var workloadHandler = workload.WorkloadHandler;
+                await workloadHandler.RemoveAsync();
+                Console.WriteLine($"Gone!");
+            }
+        }
+    }
+}
+```
+
+# 5. Remarks
+
+## 5.1. Platform Support
+
+This API is only available on Windows >= 10.0.19041.0 (aka 2004 aka 20H1).
+
+# 6. API Details
+
+```c# (but really MIDL3)
+namespace Microsoft.Windows.System.Workloads
+{
+    [contractversion(1)]
+    apicontract WorkloadsContract{};
+
+    [contract(WorkloadsContract, 1)]
+    enum WorkloadStatus
+    {
+        InProgress = 0,         // The request is in progress
+        CompletedSuccess = 1,   // The request completed successfully
+        CompletedFailure = 2,   // The request failed with some critical internal error.
+                                // See ExtendedError for HRESULT
+    };
+
+    [contract(WorkloadsContract, 1)]
+    runtimeclass WorkloadResult
+    {
+        WorkloadStatus Status { get; };
+        HRESULT ExtendedError { get; };
+    }
+
+    [contract(WorkloadsContract, 1)]
+    enum WorkloadProgressStatus
+    {
+        Queued = 0,             // The request is queued
+        InProgress = 1,         // The request is in progress
+        CompletedSuccess = 2,   // The request completed successfully
+        CompletedFailure = 3,   // The request failed with a critical internal error
+                                // See ExtendedError for HRESULT
+    };
+
+    [contract(WorkloadsContract, 1)]
+    struct WorkloadProgress
+    {
+        WorkloadProgressStatus Status;
+        Double Progress;
```

Comment:
Do we have a preferred type for Progress?  

https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.composition.animationcontroller.progress?view=windows-app-sdk-1.4#microsoft-ui-composition-animationcontroller-progress is a Single.

https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.media.imaging.downloadprogresseventargs.progress?view=windows-app-sdk-1.4 is Int.

If we use Double, we clinch the hat trick.

Reply:
PackageManager (thank you Win8) used UInt32 with values 0-100 and names it `percentage`

I snooped around a bit when writing the PackageDeploymentManager spec (https://github.com/microsoft/WindowsAppSDK/pull/4024/files line 672). Originally had "UInt32 percentage", someone mentioned "should be float" and after research determined very few use float and most use double, then during API Review feedback (Raymond's I think) was to change name from 'percentage' to 'Progress'.

@oldnewthing guidance is to use Double. Is there guidance on naming?